BeagleBone_black/src/utils/math/kalman.hpp

```diff
@@ -28,39 +28,37 @@ namespace hyped {
 namespace utils {
 namespace math {
 
-// TODO(Uday): Fill in the class comment.
-// TODO(Uday): Explain a little bit about Kalman filters in class comment or file comment
 /**
- * @brief
+ * @brief    This class is for filtering the data from sensors to smoothen it.
  *
- * @tparam T
+ * @param[in] T    This is the type of the data value that it should filter.
  */
 template <typename T>
 class Kalman {
  public:
-  // TODO(Uday): Fill in the method comments.
   /**
-   * @brief Construct a new Kalman object
+   * @brief    Construct a new Kalman object
    *
-   * @param input_value
-   * @param measurement_noise
-   * @param process_noise
-   * @param estimation_error
+   * @param[in] input_value    Initial value of the reading usually set to 0
+   * @param[in] measurement_noise    Standard deviation of the measurement
+   * @param[in] process_noise    How noisy the measurement can be (predetermined)
+   * @param[in] estimation_error    Total error that has been collected over time usually set to 0
```

Comment:
Where has the error been collected?

Reply:
Its collected overtime by filtering the data

Reply:
Then how can we input it before we start filtering?

Reply:
Yeah it isn't suppose to be there...